### PR TITLE
Upgrade dry-struct, dry-validation

### PR DESCRIPTION
**Pull request by SpComb**
Required for using the `pharos-kube-client` gem which depends on the newer version of dry-struct.

### CODE REVIEW

File: lib/pharos/addon.rb [hunk 1]
```diff
@@ -24,7 +24,8 @@ def self.addon(name, &block)
   class Addon
     include Pharos::Logging
 
-    class Schema < Dry::Validation::Schema
+    # return class as use for superclass in Dry::Validation.Params
+    Schema = Dry::Validation.Schema(build: false) do
```
Comment (SpComb) [inline lib/pharos/addon.rb]: :unamused: 

https://github.com/dry-rb/dry-validation/blob/v0.12.1/lib/dry/validation.rb#L20-L24

File: lib/pharos/configuration.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'dry-struct'
+
+module Pharos
+  module Configuration
+    class Struct < Dry::Struct
+      transform_types do |type|
+        # all attributes are optional and default to nil...
+        type.meta(omittable: true)
```
Comment (SpComb) [inline lib/pharos/configuration.rb]: I don't think we should be leaving all of these structs so weakly typed, but fixing all the optional attributes to be optional and/or fixing the specs to pass all required attributes would need work... plus duplicating more of the `ConfigSchema` `dry-validation` rules?

File: spec/pharos/phases/label_node_spec.rb [hunk 3]
```diff
@@ -102,7 +102,7 @@
             labels: { },
           },
           spec: {
-            taints: [ { key: 'node-role.kubernetes.io/master', value: nil, effect: 'NoSchedule' } ],
```
Comment (SpComb) [inline spec/pharos/phases/label_node_spec.rb]: Needs merge for #471 changes, specs are failing after master merge... I wonder why this doens't show up in drone.

File: lib/pharos/addon.rb [hunk 4]
```diff
@@ -84,7 +83,7 @@ def to_h
       end
 
       def config_schema(&block)
-        @schema = Dry::Validation.Form(Schema, &block)
+        @schema = Dry::Validation.Params(Schema, &block)
```
Comment (jakolehm) [inline lib/pharos/addon.rb]: Why `Form` -> `Params` ?

Comment (SpComb) [reply lib/pharos/addon.rb]: https://github.com/dry-rb/dry-validation/pull/407

**Pull request merged**